### ISSUE TITLE
multi-writer volume does not calculate free claims correctly

### ISSUE THREAD

**Issue by mkrueger-sabio** [hashicorp/nomad#8968]
Hi,

I registered a volume with `multi-node-multi-writer`
```
type = "csi"
id = "test-volume-httpd"
name = "test-volume-httpd"
external_id = "httpd"
access_mode = "multi-node-multi-writer"
attachment_mode = "file-system"
plugin_id = "csi.serviceware.gluster"
```

and started two jobs

```
job "httpd" {
  group "httpd" {

    restart {
      attempts = 10
      interval = "5m"
      delay    = "25s"
      mode     = "delay"
    }

    volume "htdocs" {
      type      = "csi"
      read_only = false
      source    = "test-volume-httpd"
    }

    task "httpd" {
      driver = "docker"

      volume_mount {
        volume      = "htdocs"
        destination = "/usr/local/apache2/htdocs"
        read_only   = false
      }

      config {
        image = "httpd:2.4"

        port_map {
          http = 80
        }
      }

      resources {
        network {
          port "http" {}
        }
      }
    }
  }
}
```

Of course, I could run the first job, but when I a second job I got:

```
Constraint CSI volume test-volume-httpd has exhausted its available writer claims filtered 1 node
```

My expectations was, that with `multi-node-multi-writer` I could use the volume in two jobs.

Is my expectation wrong or is this just not supported? Will this be supported in the future?

**Comment by tgross** [comment]
Hi @mkrueger-sabio! Sorry to hear about that, but thanks for reporting it!

I took a quick search through the source around that error and it looks like it's simply a bug in the [`WriteFreeClaims`](https://github.com/hashicorp/nomad/blob/v0.12.5/nomad/structs/csi.go#L384-L392) check. The `CSIVolumeAccessModeMultiNodeMultiWriter` should not be checking if there are 0 claims, but checking against the maximum allowed claims for that volume. This is admittedly not a case that's been well-tested because most of the cloud providers don't support it. Should be a small fix that I'll make sure gets out in the upcoming 0.13.

**Comment by tgross** [comment]
I've just merged #9040 and that will ship in the upcoming 0.13 release!
